libs/platform/src/lib/components/form/form.spec.ts

```diff
@@ -0,0 +1,286 @@
+        submitButton.click();
+        fixture.detectChanges();
+
+        expect(host.result).toEqual({
+            firstName: 'Tom',
+            lastName: 'Tiny',
+            favoriteColor: 'blue',
+        });
+    });
+
+});
+
+
+@Component({
+    template: `
+    <form [formGroup]="userFormGroup" (ngSubmit)="onSubmit($event)">
+        <fdp-form-group #userForm
+            [object]="user"
+            [formGroup]="userFormGroup">
+            <fdp-form-field #firstName
+                id="firstName"
+                label="First Name">
+                <fdp-input [formControl]="firstName.formControl"></fdp-input>
+            </fdp-form-field>
+            <fdp-form-field #lastName
+                id="lastName"
+                label="Last Name">
+                <fdp-input [formControl]="lastName.formControl"></fdp-input>
+            </fdp-form-field>
+            <fdp-form-field #favoriteColor
+                id="favoriteColor"
+                label="Favorite Color">
+                <fdp-input [formControl]="favoriteColor.formControl"></fdp-input>
+            </fdp-form-field>
+            <fdp-form-group #addressGroup>
+                <fdp-form-field #street
+                    id="street"
+                    label="Street">
+                    <fdp-input [formControl]="street.formControl"></fdp-input>
+                </fdp-form-field>
+                <fdp-form-field #city
+                    id="city"
+                    label="City">
+                    <fdp-input [formControl]="city.formControl"></fdp-input>
+                </fdp-form-field>
+                <fdp-form-field #state
+                    id="state"
+                    label="State">
+                    <fdp-input [formControl]="state.formControl"></fdp-input>
+                </fdp-form-field>
+            </fdp-form-group>
+        </fdp-form-group>
+        <button type="submit" #submitButton>Submit</button>
+    </form>
+    `
+})
+class NestedFormGroupsTestComponent {
+    @ViewChild('userForm') userForm: FormGroupComponent;
+    @ViewChild('firstName') firstName: FormFieldComponent;
+    @ViewChild('lastName') lastName: FormFieldComponent;
+    @ViewChild('favoriteColor') favoriteColor: FormFieldComponent;
+
+    @ViewChild('addressGroup') addressGroup: FormGroupComponent;
+    @ViewChild('street') street: FormFieldComponent;
+    @ViewChild('city') city: FormFieldComponent;
+    @ViewChild('state') state: FormFieldComponent;
+
+    @ViewChild('submitButton') submitButton: ElementRef<HTMLElement>;
+
+    public userFormGroup: FormGroup = new FormGroup({});
+
+    public user: TestUser = {
+        firstName: 'Angelica',
+        lastName: 'Mercado',
+        favoriteColor: 'red',
+        street: '123 Main St',
+        city: 'Springfield',
+        state: 'AK'
+    };
+
+    public result: any = null;
+
+    onSubmit() {
+        this.result = this.userFormGroup.value;
+    }
+
+}
+
+describe('Nested Form Groups', () => {
```

Comment:
You can better structure test so you can have describe on the main level 
```
 describe ("Form Group"
  
    describe ("core behavior"
    
    );

    describe ("layout"    
    );
  
   

    describe ("Form Fields"
        describe ("Form controls"
        )

        describe ("so on.. "
        )
     );
)
```
It is always easier to follow.

Reply:
Because FormGroup and FormFields are so integrated, I though it best to test them together  using use-case scenarios, rather than around the specific features of each. There are use cases like layout and error handling which touch on features of both components. However this is just the start of the testing of FormGroup/FormField, a lot more test need to be added, so perhaps we can discuss the structure of these test at a later time. Right now I just want to add a few tests to get started.